### PR TITLE
Fix StorageCluster check

### PR DESCRIPTION
**Pull request by petr-balogh**
Fixes: #1003

Added default timeout to 600sec for OCS deployment verification.

Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

**Comment by petr-balogh** [comment]
I see that deployment verification passed:

```
dir/openshift-cluster-dir/auth/kubeconfig get Pod  -n openshift-storage --selector=app=rook-ceph-osd -o yaml
PASSED                                                                   [100%]

- generated xml file: /home/jenkins/current-cluster-dir/logs/deployment_results.xml -
- generated html file: file:///home/jenkins/current-cluster-dir/logs/deployment_report.html -
================= 1 passed, 1 deselected in 2158.62s (0:35:58) =================
```

So marking as verified.

File: ocs_ci/ocs/ocp.py [hunk 1]
```diff
@@ -28,6 +31,9 @@ class OCP(object):
     A basic OCP object to run basic 'oc' commands
     """
 
+    # If the resource has phase set this to True in child class
```
Comment (vasukulkarni) [inline ocs_ci/ocs/ocp.py]: Did you miss the ful comment, it starts with *if* but doesn't look right.

Comment (petr-balogh) [reply ocs_ci/ocs/ocp.py]: @vasukulkarni rewrote that comment a bit, so hope that now it makes more sense

File: ocs_ci/ocs/ocp.py [hunk 2]
```diff
@@ -530,6 +537,87 @@ def check_name_is_specified(self, resource_name=''):
...
+
+        Args:
+            phase (str): Phase of resource object
+
+        Returns:
+            bool: True if phase of object is the same as passed one, False
+                otherwise.
+
+        Raises:
+            NotSupportedFunctionError: If resource doesn't have phase!
```
Comment (dahorak) [inline ocs_ci/ocs/ocp.py]: The `NotSupportedFunctionError` exception is raised from the `check_function_supported(...)` right? Then there should probably be mentioned also `ResourceNameNotSpecifiedException` raised from `check_name_is_specified()`.

File: ocs_ci/ocs/ocp.py [hunk 3]
```diff
@@ -530,6 +537,87 @@ def check_name_is_specified(self, resource_name=''):
...
+        Args:
+            support_var (bool): True if functionality is supported, False
+                otherwise.
+
+        Raises:
+            NotSupportedFunctionError: If support_var == False
+
+        """
+        if not support_var:
+            raise ResourceNameNotSpecifiedException(
```
Comment (dahorak) [inline ocs_ci/ocs/ocp.py]: Shouldn't this be `NotSupportedFunctionError`?

File: ocs_ci/ocs/ocp.py [hunk 4]
```diff
@@ -530,6 +537,87 @@ def check_name_is_specified(self, resource_name=''):
...
+
+        Args:
+            phase (str): Desired phase of resource object
+            timeout (int): Timeout in seconds to wait for desired phase
+            sleep (int): Time in seconds to sleep between attempts
+
+        Raises:
+            ResourceInUnexpectedState: In case the resource is not in expected
+                phase.
+            NotSupportedFunctionError: If resource doesn't have phase!
```
Comment (dahorak) [inline ocs_ci/ocs/ocp.py]: Same as above, the `ResourceNameNotSpecifiedException` raised from `check_name_is_specified()` should be also listed here, right?

File: ocs_ci/ocs/resources/ocs.py [hunk 5]
```diff
@@ -191,20 +196,26 @@ def ocs_install_verification(timeout=0):
...
+            resource_name=storage_cluster_name, namespace=namespace
+        )
+        log.info("Checking status of %s", storage_cluster_name)
+        log.info(
+            f"Check if StorageCluster: {local_storage_name} is in"
+            f"Succeeded phase"
         )
+        storage_cluster.wait_for_phase(
+            phase='Ready', timeout=timeout
+        ), f"StorageCluster {storage_cluster_name} not 'Ready'"
```
Comment (dahorak) [inline ocs_ci/ocs/resources/ocs.py]: I'm not sure, if I miss something, but what the string `f"StorageCluster {storage_cluster_name} not 'Ready'"` means here? Isn't it leftover from the previous `assert`?

Comment (petr-balogh) [reply ocs_ci/ocs/resources/ocs.py]: Yep, thanks for catch but hopefully it didn't cause any issue as I saw that verification passed. DONE

**Comment by petr-balogh** [comment]
Ran verification once more with production deviceset here: https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/2192/console

Taking approve from Vasu and merging. Thanks

**Pull request merged**